src/modules/Builder/features/ActivitySettings/ScoresAndReports/ScoreContent/ScoreContent.tsx

```diff
@@ -0,0 +1,165 @@
+
+  useEffect(() => {
+    isScoreIdVariable && setIsChangeScoreIdPopupVisible(true);
+  }, [isScoreIdVariable, scoreName]);
+
+  useEffect(() => setValue(`${name}.id`, scoreId), [scoreName]);
+
+  useEffect(() => {
+    const selectedItems = items.filter((item) => itemsScore.includes(item.id as string));
+    const { minScore, maxScore } = getScoreRange(selectedItems, calculationType);
+    setScoreRangeLabel(getScoreRangeLabel(minScore, maxScore));
+    setValue(`${name}.minScore`, minScore);
+    setValue(`${name}.maxScore`, maxScore);
+  }, [itemsScore, calculationType]);
+
+  const copyScoreId = () => {
+    navigator.clipboard.writeText(scoreId);
+  };
+
+  const onChangeScoreId = () => {
+    setValue(`${name}.id`, scoreId);
+  };
+
+  return (
+    <StyledFlexColumn>
+      <StyledFlexTopStart>
+        <Box sx={{ mr: theme.spacing(4.8), width: '50%' }}>
+          <InputController
+            name={`${name}.name`}
+            label={t('scoreName')}
+            sx={{ mb: theme.spacing(4.8) }}
+          />
+          <SelectController
+            name={`${name}.calculationType`}
+            sx={{ mb: theme.spacing(4.8) }}
+            control={control}
+            options={calculationTypes}
+            label={t('scoreCalculationType')}
+            fullWidth
+          />
+        </Box>
+        <Box>
+          <StyledTitleSmall>{t('scoreId')}</StyledTitleSmall>
+          <StyledFlexTopCenter>
+            <StyledBodyLarge>{scoreId}</StyledBodyLarge>
+            <StyledDuplicateButton
+              sx={{ p: theme.spacing(1), mr: theme.spacing(0.2) }}
+              onClick={copyScoreId}
+            >
+              <Svg id="duplicate" width="20" height="20" />
+            </StyledDuplicateButton>
+          </StyledFlexTopCenter>
+          <StyledTitleSmall sx={{ mb: theme.spacing(1.2) }}>{t('rangeOfScores')}</StyledTitleSmall>
+          <StyledBodyLarge sx={{ mb: theme.spacing(2.4) }}>{scoreRangeLabel}</StyledBodyLarge>
+        </Box>
+      </StyledFlexTopStart>
+      <StyledTitleMedium>{t('scoreItems')}</StyledTitleMedium>
+      <StyledFlexTopStart sx={{ m: theme.spacing(1.2, 0, 4.4, 0), alignItems: 'flex-end' }}>
+        <TransferListController
+          name={`${name}.itemsScore`}
+          items={tableItems}
+          columns={scoreItemsColumns}
+          selectedItemsColumns={selectedItemsColumns}
+          hasSelectedSection
+          hasSearch
+        />
+      </StyledFlexTopStart>
+      <Switch
+        name={`${name}.showMessage`}
+        control={control}
+        label={t('showMessage')}
+        tooltipText={t('showMessageTooltip')}
+      />
+      {showMessage && (
+        <StyledEditor uiType={EditorUiType.Secondary} name={`${name}.message`} control={control} />
+      )}
+      <Switch
+        name={`${name}.printItems`}
+        control={control}
+        label={t('printItems')}
+        tooltipText={t('printItemsTooltip')}
+      />
+      {printItems && (
```

Comment:
is it correct to have condition with `printItems`, not with `items` which is used within `TransferListController`?

Reply:
printItems - boolean. and in TransferListController related to itemsPrint: Item[]

Reply:
isPrintItems?

Reply:
your variants are better, but backend uses such naming